From a24d76afa4f0dc4528ecc9e1cd8826951a41c6e5 Mon Sep 17 00:00:00 2001
From: Vamsi Krishna Brahmajosyula <vbrahmajosyula@vmware.com>
Date: Tue, 11 Oct 2022 00:09:35 +0530
Subject: [PATCH] aarch64: Fix ld RWX warning

The current ld script places .dynamic section right after .text, which
results in that one LOAD segment contains both data and code. Binutils
2.39 and later will throw a warning if RWX segments exist. Code and
data should be placed in different LOAD segments with different
permissions. This patch adds a gap between data and code sections which
is taken from the default ld script in riscv64.

Ported from upstream, patch for riscv64
[upstream 98f1250742bbe3d91a7aaf5d66cc2cb4bdb3c0dd]

Co-developed-by: Moody Liu <mooodyhunter@outlook.com>
Signed-off-by: Moody Liu <mooodyhunter@outlook.com>
Signed-off-by: Letu Ren <fantasquex@gmail.com>
Signed-off-by: Vamsi Krishna Brahmajosyula <vbrahmajosyula@vmware.com>
---
 gnuefi/elf_aarch64_efi.lds | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gnuefi/elf_aarch64_efi.lds b/gnuefi/elf_aarch64_efi.lds
index 836d982..f08f403 100644
--- a/gnuefi/elf_aarch64_efi.lds
+++ b/gnuefi/elf_aarch64_efi.lds
@@ -15,6 +15,7 @@ SECTIONS
   }
   _etext = .;
   _text_size = . - _text;
+  . = DATA_SEGMENT_ALIGN (CONSTANT (MAXPAGESIZE), CONSTANT (COMMONPAGESIZE));
   .dynamic  : { *(.dynamic) }
   .data : ALIGN(4096)
   {
-- 
2.37.3

